### PR TITLE
Blocks: short-circuit early on sites that do not support blocks

### LINKED ISSUE #430: Plugin breaks ClassicPress
Yesterday (2023-09-13) I upgraded the plugin in my ClassicPress installation on blog.atari-frosch.de, and directly afterwards the blog only delivered empty pages. I had to delete the plugin on the shell (I'm hosting the stuff myself) so that the blog could start working correctly again.

System: Devuan GNU/Linux 4 Chimaera 
Kernel: Linux seelilie 5.10.0-25-amd64 #1 SMP Debian 5.10.191-1 (2023-08-16) x86_64 GNU/Linux
Webserver:  nginx-full_1.18.0-6.1+deb11u3_all.deb 
PHP: PHP 7.4.33 with php7.4-fpm_7.4.33-1+deb11u3_amd64.deb

This is what the error.log said after the update (2a01:4f8:150:81e4::2 is the server's IPv6 address):

2023/09/13 17:02:24 [error] 17262#17262: *694892 FastCGI sent in stderr: "Primary script unknown" while reading response
 header from upstream, client: 178.63.18.4, server: blog.atari-frosch.de, request: "GET /status.php HTTP/1.1", upstream:
 "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de"
2023/09/13 17:02:37 [error] 17262#17262: *694964 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-cron.php(34): require_once('/srv/atarifrosc...')
#9 {main}
  thrown in /srv/atarifrosch-blo" while reading response header from upstream, client: 2a01:4f8:150:81e4::2, server: blog.atari-frosch.de, request: "POST /wp-cron.php?doing_wp_cron=1694617357.0145969390869140625000 HTTP/1.1", upstream: "fas
tcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de", referrer: "https://blog.atari-frosch.de/wp-cron.php?doing_wp_cron=1694617357.0145969390869140625000"
2023/09/13 17:02:37 [error] 17262#17262: *694962 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-admin/admin-ajax.php(22): require_once('/srv/atarifrosc...')
#9 {main}
  thrown in /srv/ata" while reading response header from upstream, client: 44.235.211.232, server: blog.atari-frosch.de,
 request: "GET /wp-admin/admin-ajax.php?action=wordfence_doScan&isFork=0&scanMode=quick&cronKey=92986d278e1731d7d9c6382dadc82053&signature=d074f3ceead03583d4a5ea1e2ecc2f70375108a93e089b1b37a2b00ce487fde8&remote=1 HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de"
2023/09/13 17:02:38 [error] 17262#17262: *694967 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-admin/admin-ajax.php(22): require_once('/srv/atarifrosc...')
#9 {main}
  thrown in /srv/ata" while reading response header from upstream, client: 44.235.211.232, server: blog.atari-frosch.de,
 request: "GET /wp-admin/admin-ajax.php?action=wordfence_doScan&isFork=0&scanMode=quick&cronKey=92986d278e1731d7d9c6382dadc82053&signature=d074f3ceead03583d4a5ea1e2ecc2f70375108a93e089b1b37a2b00ce487fde8&remote=1 HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de"
2023/09/13 17:02:42 [error] 17262#17262: *694935 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-admin/admin-ajax.php(22): require_once('/srv/atarifrosc...')
#9 {main}
  thrown in /srv/ata" while reading response header from upstream, client: 109.90.207.33, server: blog.atari-frosch.de,
request: "POST /wp-admin/admin-ajax.php HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.ata
ri-frosch.de", referrer: "https://blog.atari-frosch.de/wp-admin/plugins.php"
2023/09/13 17:02:54 [error] 17262#17262: *695008 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-admin/admin.php(34): require_once('/srv/atarifrosc...')
#9 /srv/atarifrosch-blog/wp-admin/p" while reading response header from upstream, client: 109.90.207.33, server: blog.atari-frosch.de, request: "GET /wp-admin/plugins.php HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de", referrer: "https://blog.atari-frosch.de/wp-admin/plugins.php"
2023/09/13 17:03:14 [error] 17262#17262: *695060 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-blog-header.php(13): require_once('/srv/atarifrosc...')
#9 /srv/atarifrosch-blog/index.php(" while reading response header from upstream, client: 109.90.207.33, server: blog.atari-frosch.de, request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de"
2023/09/13 17:03:39 [error] 17262#17262: *695104 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-cron.php(34): require_once('/srv/atarifrosc...')
#9 {main}
  thrown in /srv/atarifrosch-blo" while reading response header from upstream, client: 2a01:4f8:150:81e4::2, server: blog.atari-frosch.de, request: "POST /wp-cron.php?doing_wp_cron=1694617419.0569748878479003906250 HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de", referrer: "https://blog.atari-frosch.de/wp-cron.php?doing_wp_cron=1694617419.0569748878479003906250"
2023/09/13 17:03:39 [error] 17262#17262: *695100 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-blog-header.php(13): require_once('/srv/atarifrosc...')
#9 /srv/atarifrosch-blog/index.php(" while reading response header from upstream, client: 162.19.87.99, server: blog.atari-frosch.de, request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de"
2023/09/13 17:03:39 [error] 17262#17262: *695106 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-cron.php(34): require_once('/srv/atarifrosc...')
#9 {main}
  thrown in /srv/atarifrosch-blo" while reading response header from upstream, client: 2a01:4f8:150:81e4::2, server: blog.atari-frosch.de, request: "POST /wp-cron.php?doing_wp_cron=1694617419.0634241104125976562500 HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de", referrer: "https://blog.atari-frosch.de/wp-cron.php?doing_wp_cron=1694617419.0634241104125976562500"
2023/09/13 17:03:39 [error] 17262#17262: *695101 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-blog-header.php(13): require_once('/srv/atarifrosc...')
#9 /srv/atarifrosch-blog/index.php(" while reading response header from upstream, client: 35.159.31.152, server: blog.atari-frosch.de, request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de"
2023/09/13 17:03:41 [error] 17262#17262: *695115 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-blog-header.php(13): require_once('/srv/atarifrosc...')
#9 /srv/atarifrosch-blog/index.php(" while reading response header from upstream, client: 116.203.72.98, server: blog.atari-frosch.de, request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de"
2023/09/13 17:03:43 [error] 17262#17262: *695121 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-blog-header.php(13): require_once('/srv/atarifrosc...')
#9 /srv/atarifrosch-blog/index.php(" while reading response header from upstream, client: 152.228.220.143, server: blog.atari-frosch.de, request: "GET /tag/terrorismus/feed/ HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de"
2023/09/13 17:04:50 [error] 17262#17262: *695134 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-cron.php(34): require_once('/srv/atarifrosc...')
#9 {main}
  thrown in /srv/atarifrosch-blo" while reading response header from upstream, client: 2a01:4f8:150:81e4::2, server: blog.atari-frosch.de, request: "POST /wp-cron.php?doing_wp_cron=1694617490.5343809127807617187500 HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de", referrer: "https://blog.atari-frosch.de/wp-cron.php?doing_wp_cron=1694617490.5343809127807617187500"
2023/09/13 17:04:50 [error] 17262#17262: *695132 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-blog-header.php(13): require_once('/srv/atarifrosc...')
#9 /srv/atarifrosch-blog/index.php(" while reading response header from upstream, client: 92.247.181.9, server: blog.atari-frosch.de, request: "GET /feed/ HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de", referrer: "https://blog.atari-frosch.de/"
2023/09/13 17:04:50 [error] 17262#17262: *695137 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_bloc
ks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-blog-header.php(13): require_once('/srv/atarifrosc...')
#9 /srv/atarifrosch-blog/index.php(" while reading response header from upstream, client: 152.228.220.143, server: blog.
atari-frosch.de, request: "GET /2011/05/02/merkel-lobt-obama-fuer-einen-mord/feed/ HTTP/1.1", upstream: "fastcgi://unix:
/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de"
2023/09/13 17:05:10 [error] 17262#17262: *695141 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-blog-header.php(13): require_once('/srv/atarifrosc...')
#9 /srv/atarifrosch-blog/index.php(" while reading response header from upstream, client: 195.62.45.130, server: blog.atari-frosch.de, request: "GET /feed/ HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de"
2023/09/13 17:05:24 [error] 17262#17262: *695150 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-blog-header.php(13): require_once('/srv/atarifrosc...')
#9 /srv/atarifrosch-blog/index.php(" while reading response header from upstream, client: 195.62.45.130, server: blog.atari-frosch.de, request: "GET /feed/ HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de"
2023/09/13 17:05:42 [error] 17262#17262: *695156 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error:
 Call to undefined function register_block_type_from_metadata() in /srv/atarifrosch-blog/wp-content/plugins/activitypub/
includes/class-blocks.php:32
Stack trace:
#0 /srv/atarifrosch-blog/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\Blocks::register_blocks()
#1 /srv/atarifrosch-blog/wp-content/plugins/activitypub/activitypub.php(55): Activitypub\Blocks::init()
#2 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(287): Activitypub\init()
#3 /srv/atarifrosch-blog/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /srv/atarifrosch-blog/wp-includes/plugin.php(458): WP_Hook->do_action()
#5 /srv/atarifrosch-blog/wp-settings.php(453): do_action()
#6 /srv/atarifrosch-blog/wp-config.php(98): require_once('/srv/atarifrosc...')
#7 /srv/atarifrosch-blog/wp-load.php(37): require_once('/srv/atarifrosc...')
#8 /srv/atarifrosch-blog/wp-blog-header.php(13): require_once('/srv/atarifrosc...')
#9 /srv/atarifrosch-blog/index.php(" while reading response header from upstream, client: 185.220.102.241, server: blog.atari-frosch.de, request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "blog.atari-frosch.de"

Before the upgrade the plugin didn't work. I could never find my instance in the Fediverse.

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #430

## Proposed changes:

This is typically only the case for sites using a custom version of WordPress, like ClassicPress.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:

> **Note**
> You will need a site using ClassicPress to test this.

1. Install ActivityPub version 1.0.0.
2. Notice the Fatal Error in your logs.
3. Switch to this branch.
4. The Fatal Error should disappear.

### CODE REVIEW

File: includes/class-blocks.php [hunk 1]
```diff
@@ -6,6 +6,11 @@
 
 class Blocks {
 	public static function init() {
+		// Short-circuit early if the site does not support blocks.
+		if ( ! \function_exists( 'register_block_type_from_metadata' ) ) {
```
Comment (pfefferle) [inline includes/class-blocks.php:10]: Should we add a function to check for block editor support? Something like `are_blocks_supported` and add also a filter?

Comment (pfefferle) [reply includes/class-blocks.php:10]: And maybe add the check to the main activitypub.php file, that would save a class init!

**Comment by Atari-Frosch** [comment]
I changed the first block in the mentioned file (no, I do not really know PHP,  „my“ language is Python) to this:

class Blocks {
        public static function init() {
                // Short-circuit early if the site does not support blocks.
                if ( ! \function_exists( 'register_block_type_from_metadata' ) ) {
                        // this is already being called on the init hook, so just add it.
                        self::register_blocks();
                        \add_action( 'wp_enqueue_scripts', array( self::class, 'add_data' ) );
                        \add_action( 'enqueue_block_editor_assets', array( self::class, 'add_data' ) );
                }
        }

– adding a closing curly bracket to close the if-statement.

Well. No change. Pages are empty.

**Comment by pfefferle** [comment]
@Atari-Frosch if you add it like you did, you have to remove the `!` in the if statement!

**Comment by Atari-Frosch** [comment]
> @Atari-Frosch if you add it like you did, you have to remove the `!` in the if statement!

That was it. Thank you! Now at least it doesn't break the site anymore. Now let's see if I can find my blog in the Fediverse after my next blog article. :-)

File: includes/class-blocks.php (hunk 1 above)
Comment (jeherve) [reply includes/class-blocks.php]: All good ideas, I've just done that. 👍

**Review by pfefferle — APPROVED** [review]
thanks @jeherve

**Pull request merged**